core/src/test/java/com/datastax/oss/driver/internal/core/type/codec/registry/CachingCodecRegistryTest.java

```diff
@@ -758,58 +298,47 @@ public void should_create_tuple_codec_for_cql_type() {
   }
 
   @Test
-  public void should_create_tuple_codec_for_cql_type_and_java_value() {
-    TupleType cqlType = DataTypes.tupleOf(DataTypes.INT, DataTypes.listOf(DataTypes.TEXT));
-    TupleValue value = cqlType.newValue();
-
+  @UseDataProvider(
+      value = "tuplesWithCqlTypes",
+      location = CachingCodecRegistryTestDataProviders.class)
+  public void should_create_tuple_codec_for_cql_type_and_java_value(
+      DataType cqlType, Object value) {
     TestCachingCodecRegistry registry = new TestCachingCodecRegistry(mockCache);
     InOrder inOrder = inOrder(mockCache);
-
-    TypeCodec<TupleValue> codec = registry.codecFor(cqlType, value);
+    TypeCodec<?> codec = registry.codecFor(cqlType, value);
     assertThat(codec).isNotNull();
     assertThat(codec.accepts(cqlType)).isTrue();
     assertThat(codec.accepts(GenericType.TUPLE_VALUE)).isTrue();
     assertThat(codec.accepts(TupleValue.class)).isTrue();
     assertThat(codec.accepts(value)).isTrue();
-    inOrder.verify(mockCache).lookup(cqlType, GenericType.TUPLE_VALUE, false);
-
+    inOrder.verify(mockCache).lookup(cqlType, GenericType.of(DefaultTupleValue.class), true);
```

Comment:
I think that `isJavaCovariant` should always be true when we are creating a codec for an actual object. The fact that `isJavaCovariant` was set to `false` for instances of `TupleValue` and `UdtValue` was somewhat artificial, and only possible because these two interfaces were being special-cased [here](https://github.com/datastax/java-driver/commit/8e4464e1634a97ab2e25a8de256be893d1cc24bd#diff-34d4c85a5bfd4f5ccb02c01deb989c99L246) and [here](https://github.com/datastax/java-driver/commit/8e4464e1634a97ab2e25a8de256be893d1cc24bd#diff-34d4c85a5bfd4f5ccb02c01deb989c99L276-L280).

Reply:
Agreed, this was a side effect of the previous implementation.